google/cloud/security/inventory/pipelines/load_projects_buckets_acls_pipeline.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pipeline to load bucket acls into Inventory."""
+
+import json
+
+# pylint: disable=line-too-long
+from google.cloud.security.common.data_access import errors as data_access_errors
+from google.cloud.security.common.gcp_api import errors as api_errors
+from google.cloud.security.common.util import log_util
+from google.cloud.security.inventory import errors as inventory_errors
+from google.cloud.security.inventory.pipelines import base_pipeline
+# pylint: enable=line-too-long
+
+
+LOGGER = log_util.get_logger(__name__)
+
+
+class LoadProjectsBucketsAclsPipeline(base_pipeline.BasePipeline):
+    """Pipeline to load project buckets data into Inventory."""
+
+    PROJECTS_RESOURCE_NAME = 'project_iam_policies'
+    RESOURCE_NAME = 'buckets_acl'
+
+    def __init__(self, cycle_timestamp, configs, gcs_client, dao):
+        """Constructor for the data pipeline.
+
+        Args:
+            cycle_timestamp: String of timestamp, formatted as YYYYMMDDTHHMMSSZ.
+            configs: Dictionary of configurations.
+            gcs_client: GCS API client.
+            dao: Data access object.
+
+        Returns:
+            None
+        """
+        super(LoadProjectsBucketsAclsPipeline, self).__init__(
+            cycle_timestamp, configs, gcs_client, dao)
+
+    def _transform(self, buckets_acls_maps):
+        """Yield an iterator of loadable bucket acls.
+
+        Args:
+            buckets_acls_maps: An iterable of bucket acls as per-bucket
+                dictionary.
+                Example: {'bucket_name': 'axample_bucket_name.appspot.com',
+                          'acl': bucket_acls_json}
+
+        Yields:
+            An iterable of bucket acls, as a per-bucket dictionary.
+        """
+        for buckets_map_acls in buckets_acls_maps:
+            acls = buckets_map_acls['acl']
```

Comment:
Looking at the retrieve function below, an error will be logged if no acl is found. 
 So, buckets_maps_acls['acl'] should always be populated at this point.  One nit to fix is to make the variable naming consistent:
```buckets_maps_acls  -> buckets_acls_map```
Done.